### PR TITLE
Renamed RICE coding to Exponential Golomb coding

### PR DESCRIPTION
**Pull request by bumblebritches57**
I feel like some of the wording could use a bit of work, but it's more or less good to go.

### CODE REVIEW

File: flac.md [hunk 1]
```diff
@@ -498,49 +498,49 @@ Data      | Description
...
 `u(2)`     | `RESIDUAL_CODING_METHOD`
-`RESIDUAL_CODING_METHOD_PARTITIONED_RICE` \|\| `RESIDUAL_CODING_METHOD_PARTITIONED_RICE2` |
+`RESIDUAL_CODING_METHOD_PARTITIONED_Exponential-Golomb` \|\| `RESIDUAL_CODING_METHOD_PARTITIONED_Exponential-Golomb2` |
 
 ### RESIDUAL_CODING_METHOD
 Value | Description
 -----:|:-----------
-00    | partitioned Rice coding with 4-bit Rice parameter; RESIDUAL_CODING_METHOD_PARTITIONED_RICE follows
-01    | partitioned Rice coding with 5-bit Rice parameter; RESIDUAL_CODING_METHOD_PARTITIONED_RICE2 follows
+00    | partitioned Exponential-Golomb coding with 4-bit Exponential-Golomb parameter; RESIDUAL_CODING_METHOD_PARTITIONED_Exponential-Golomb follows
```
Comment (dericed) [inline flac.md]: should be continue with all caps?

Comment (bumblebritches57) [inline flac.md]: That was just a quick find and replace, I'll shorten it to just exp-golomb in paragraphs, one sec.

**Comment by dericed** [comment]
IIUC Rice coding is a subset of Golomb coding. The spec acknowledges Robert Rice for the coding used in FLAC. Is this patch intended to say that FLAC uses Golomb coding but not the Rice subset?

Would it be more complete to say `Exponential Golomb Rice coding`?

Also do you have a citation for Exponential-Golomb?

**Comment by bumblebritches57** [comment]
`Exponential Golomb Rice` would actually be a better way to word it.

As for the citation I'm not sure what you mean? the description of the coding process exactly matches that of expoential-golomb coding.

File: flac.md [hunk 2]
```diff
@@ -84,11 +84,11 @@ FLAC uses four methods for modeling the input signal:
 
 FLAC currently defines two similar methods for the coding of the error signal from the prediction stage. The error signal is coded using Exponential-Golomb codes in one of two ways:
 
-1. the encoder estimates a single Exponential-Golomb parameter based on the variance of the residual and Exponential-Golomb codes the entire residual using this parameter;
+1. the encoder estimates a single exp-golomb parameter based on the variance of the residual and exp-golomb codes the entire residual using this parameter;
```
Comment (dericed) [inline flac.md]: alternatively, perhaps we should start a definition section at the beginning (such as this one https://github.com/FFmpeg/FFV1/blob/master/ffv1.md#definitions) and define `exp-golomb` more explicitly there before using the term.

Comment (bumblebritches57) [inline flac.md]: Makes sense, I'll go ahead and do that as well.

**Comment by dericed** [comment]
I mean a citation for those who want to learn more about Exponential Golomb Rice coding. The specification is not self-sufficient in that the reader needs to understand Exponential Golomb Rice coding in order to fully understand FLAC, so we should add a normative reference for it.

**Comment by bumblebritches57** [comment]
There's a wiki page on it, and I'll link that in the introduction, which I've put in the residual coding section.

https://en.wikipedia.org/wiki/Exponential-Golomb_coding

File: flac.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ Many terms like "block" and "frame" are used to mean different things in differe
 
 - **Subframe**: A subframe header plus one or more encoded samples from a given channel. All subframes within a frame will contain the same number of samples.
 
+- ##Exponential-Golomb coding##: One of Robert Rice's universal coding schemes, FLAC's residual coder, compresses data by writing the number of bits to be read minus 1, before writing the actual value
```
Comment (retokromer) [inline flac.md]: Please replace `##` by `** ` and add a period at the end.

File: flac.md [hunk 4]
```diff
@@ -82,13 +84,15 @@ FLAC uses four methods for modeling the input signal:
 
 # Residual Coding
 
-FLAC currently defines two similar methods for the coding of the error signal from the prediction stage. The error signal is coded using Rice codes in one of two ways:
+FLAC uses Exponential-Golomb (a variant of Rice) coding as it's residual encoder. [You can learn more about Exp-Golomb coding on Wikipedia](https://en.wikipedia.org/wiki/Exponential-Golomb_coding)
```
Comment (retokromer) [inline flac.md]: I would say that exp-golomb is Exponential-Golomb (and not capitalise the «abbreviation»).

**Comment by dericed** [comment]
Is this PR intended to be open?

**Comment by retokromer** [comment]
In my opinion it was the one that superseded yours, and has not been merged so far.

**Comment by dericed** [comment]
My PR was a minor single copy/paste whereas this was represents some actual work. @bumblebritches57 did you intend for this to be closed without merging?

**Comment by bumblebritches57** [comment]
@dericed No, I did not. I got a message saying it was accepted, and thought that meant it was merged in, and obviously it was not.

Sorry for the confusion.

**Comment by dericed** [comment]
Oh I think you say that @retokromer `approved` the request at https://github.com/privatezero/flac_markdown/pull/39#pullrequestreview-39851517. This is simply a process for expressing consensus about it. It hasn't been accepted/merged yet since it was closed before we could do that.

**Comment by dericed** [comment]
Hi @bumblebritches57, because we merged #38 this pr now has a conflict. You can resolve that in the website via `Resolve conflicts` before we can review. Let us know if you need help with that or I could do it for you.

**Comment by bumblebritches57** [comment]
I'm not really seeing any problem, the only difference I see is the 0b stuff.

When I made these 2 branches, I based them off master, should I instead have made them all in one branch, or based this branch on the other one?

**Comment by bumblebritches57** [comment]
Ok, I just pushed changed to this branch to fix the merge conflicts.

**Comment by privatezero** [comment]
There were a few more conflicts that I just took a stab at editing - if people would give this a last glance then I will go ahead and merge!

File: flac.md [hunk 5]
```diff
@@ -498,49 +502,49 @@ Data      | Description
...
 Value       | Description
 -----------:|:-----------
-0b00        | partitioned Rice coding with 4-bit Rice parameter; RESIDUAL_CODING_METHOD_PARTITIONED_RICE follows
-0b01        | partitioned Rice coding with 5-bit Rice parameter; RESIDUAL_CODING_METHOD_PARTITIONED_RICE2 follows
+0b00        | partitioned Exp-Golomb coding with 4-bit Exp-Golomb parameter; RESIDUAL_CODING_METHOD_PARTITIONED_EXPGOLOMB follows
+0b01        | partitioned Exp-Golomb coding with 5-bit Exp-Golomb parameter; RESIDUAL_CODING_METHOD_PARTITIONED_EXPGOLOMB2 follows
 0b10 - 0b11 | reserved
 
-### RESIDUAL_CODING_METHOD_PARTITIONED_RICE
+### RESIDUAL_CODING_METHOD_PARTITIONED_EXPGOLOMB
```
Comment (retokromer) [inline flac.md]: I suggest `RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB` for readability.

File: flac.md [hunk 6]
```diff
@@ -498,49 +502,49 @@ Data      | Description
...
-#### RICE_PARTITION
+#### EXP_GOLOMB_PARTITION
 Data       | Description
 :----------|:-----------
-`u(4(+5))` | `RICE PARTITION ENCODING PARAMETER` (see [section on `RICE PARTITION ENCODING PARAMETER`](#rice-partition-encoding-parameter))
+`u(4(+5))` | `EXP-GOLOMB PARTITION ENCODING PARAMETER` (see [section on `EXP-GOLOMB PARTITION ENCODING PARAMETER`](#Exponential-Golomb-partition-encoding-parameter))
 `u(?)`     | `ENCODED RESIDUAL` (see [section on `ENCODED RESIDUAL`](#encoded-residual))
 
-#### RICE PARTITION ENCODING PARAMETER
+#### EXPGOLOMB PARTITION ENCODING PARAMETER
```
Comment (retokromer) [inline flac.md]: I suggest `EXP-GOLOMB PARTITION ENCODING PARAMETER` for readability.

File: flac.md [hunk 7]
```diff
@@ -498,49 +502,49 @@ Data      | Description
...
 Value           | Description
 ---------------:|:-----------
-0b0000 - 0b1110 | Rice parameter.
+0b0000 - 0b1110 | Exp-golomb parameter.
 0b1111          | Escape code, meaning the partition is in unencoded binary form using n bits per sample; n follows as a 5-bit number.
 
-### RESIDUAL_CODING_METHOD_PARTITIONED_RICE2
+### RESIDUAL_CODING_METHOD_PARTITIONED_EXPGOLOMB2
```
Comment (retokromer) [inline flac.md]: I suggest `RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB2` for readability.

File: flac.md [hunk 8]
```diff
@@ -498,49 +502,49 @@ Data      | Description
...
-#### RICE2_PARTITION
+#### EXP_GOLOMB2_PARTITION
 Data       | Description
 :----------|:-----------
-`u(5(+5))` | `RICE2 PARTITION ENCODING PARAMETER` (see [section on `RICE2 PARTITION ENCODING PARAMETER`](#rice2-partition-encoding-parameter))
+`u(5(+5))` | `EXPGOLOMB2 PARTITION ENCODING PARAMETER` (see [section on `EXPGOLOMB2 PARTITION ENCODING PARAMETER`](#expgolomb2-partition-encoding-parameter))
 `u(?)`     | `ENCODED RESIDUAL` (see [section on `ENCODED RESIDUAL`](#encoded-residual))
 
-#### RICE2 PARTITION ENCODING PARAMETER
+#### EXPGOLOMB2 PARTITION ENCODING PARAMETER
```
Comment (retokromer) [inline flac.md]: I suggest `EXP-GOLOMB2 PARTITION ENCODING PARAMETER ` for readability.

Comment (bumblebritches57) [inline flac.md]: I'll make these changes, 1 sec.

**Comment by retokromer** [comment]
@bumblebritches57 Please wait until we merged, before deleting. Thank you! And your merged contributions will, of course, be visible as https://github.com/privatezero/flac_markdown/commits?author=bumblebritches57.

**Comment by retokromer** [comment]
@privatezero In my opinion, we can now squash and merge. Shall I do so?

**Comment by privatezero** [comment]
Sounds good @retokromer

**Comment by retokromer** [comment]
@privatezero If needed, we can always modify.

@bumblebritches57 Thank you very much for your contribution! You can now delete your branch.

**Comment by privatezero** [comment]
Thanks @bumblebritches57 !

**Pull request merged**